src/persist/src/indexed/encoding.rs

```diff
@@ -476,6 +476,20 @@ impl BlobMeta {
         }
         Ok(())
     }
+
+    /// Current encoding version that BlobMeta can be read and written as.
+    pub fn version() -> u8 {
+        BlobMeta::CURRENT_VERSION
+    }
+
+    /// Hint for the encoded version of this previously encoded BlobMeta.
+    ///
+    /// Returns None if it is not possible to determine the encoding version.
+    pub fn encoded_version(buf: &[u8]) -> Option<u8> {
```

Comment:
nit: I'd have this be a Result with the error you currently have on the callsite